kasa/protocols/smartprotocol.py

```diff
@@ -285,7 +287,9 @@ async def _execute_multiple_query(self, requests: dict, retry_count: int) -> dic
                 resp = await self._transport.send(
                     self.get_smart_request(method, params)
                 )
-                self._handle_response_error_code(resp, method, raise_on_error=False)
+                self._handle_response_error_code(
+                    resp, method, raise_on_error=raise_on_error
+                )
```

Comment:
With `FORCE_SINGLE_REQUEST` this also needs to use the `raise_on_error` variable